pkg/event/sqlite.go

```diff
@@ -129,17 +135,26 @@ func (repo SQLiteRepository) ListResourceEvents(ctx context.Context, listParams
 		CreatedAt: lastResourceEvent.CreatedAt,
 	})
 	if err != nil {
-		return resourceEvents, "", err
+		return models, "", err
+	}
+
+	for i := range resourceEvents {
```

Comment:
This loop should be moved above line 128 so that in the scenario that the result set is smaller than the specified `Limit`, the appropriate result is still returned.